src/Testing/src/AssemblyTestLog.cs

```diff
@@ -238,10 +256,19 @@ public static AssemblyTestLog ForAssembly(Assembly assembly)
                     catch { }
                 }
             }
+
             return log;
         }
     }
 
+    public void ReportTestFailure()
+    {
+        if (_cleanupState == CleanupState.CleanupRequested)
```

Comment:
While I get what this is doing, the check isn't really needed and sort of makes it confusing when reading.
I would just always set the state, change the function name to something like "MarkHasTestFailure" or "RecordHasTestFailure", and if you really like the check maybe throw an exception.

Reply:
Glad you pointed me here @BrennanConroy. My intention was to ensure we didn't lose track of reported test failures because another class requested cleanup _later_. This line means problems reported before the first cleanup request are ignored ☹️

Just updating `_cleanupState` is more correct but insufficient because other `AssemblyTestLog` users in the same assembly might not inherit from `LoggedTestBase`. I need to ensure all users report their failures if any users exist in the assembly because the cleanup has to be all or nothing.

Reply:
I am working on two options here:

1. Revert changes to `LoggedTestBase` and have `AspNetTestAssemblyRunner` somehow notify `AssemblyTestLog` of test failures. This would entirely eliminate the concept of registering for cleanup and would mean _any_ failure in a test assembly prevents cleanup.
2. Update all `AssemblyTestLog` users to make sure they notify that class of test failures. Basically this option would make _not_ registering fail loudly and longly. If I go this route, I'll use @halter73's two-`bool` suggestion.